pkg/driver/node_server.go

```diff
@@ -498,7 +503,51 @@ func (driver *Driver) stageVolume(
 
 	// Store mount info in the staging device
 	stagingDevice.MountInfo = mountInfo
+	
+	// CON-3010
+	// Get the PVC details to identify the datasource of PVC
+	log.Infof("PVC Name %s and PVC Namespace %s", volumeContext[pvcNameAttribute], volumeContext[pvcNamespaceAttribute])
+	
+	if volumeContext[pvcNameAttribute] != "" && volumeContext[pvcNamespaceAttribute] != "" {
+		pvc, err := driver.flavor.GetPVCByName( volumeContext[pvcNameAttribute], volumeContext[pvcNamespaceAttribute] )
+		if err != nil{
+			return nil, status.Error(codes.Internal, fmt.Sprintf("Error getting pvc data source for volume %v, %v", volumeID, err))
+		}
+		if pvc.Spec.DataSource != nil {
 
+			// If the DataSource is a VolumeSnapshot
+			if pvc.Spec.DataSource.Kind == "VolumeSnapshot" {
+				log.Infof("  Source Kind: VolumeSnapshot \n VolumeSnapshot Name: %s ", pvc.Spec.DataSource.Name)
```

Comment:
You can make this trace log as it is not readable. If info is required, then modify the comment accordingly.